proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,386 @@
+JavaScript and styles, and use the `main` or `module` root keys to define the
+JavaScript entry point. Some packages use the `"sass"` key at the root of their
+`package.json`.
+
+Other packages have adopted [conditional exports], driven by build tools like
+[Vite], [Parcel] and [Sass Loader for Webpack] which all resolve Sass paths
+using the `"sass"` and the `"style"` custom conditions.
+
+[conditional exports]: https://nodejs.org/api/packages.html#conditional-exports
+[Vite]: https://github.com/vitejs/vite/pull/7817
+[Parcel]:
+    https://github.com/parcel-bundler/parcel/blob/2d2400ded4615375ee6bd53ef77b4857ad1591dd/packages/transformers/sass/src/SassTransformer.js#L163
+[Sass Loader for Webpack]:
+    https://github.com/webpack-contrib/sass-loader/blob/02df41203adfda96959e56abb43bd35a89ec11ba/src/utils.js#L514
+
+Because use of conditional exports is flexible and recommended for modern
+packages, this will be the primary method used for the Node package importer. We
+will support both the `"sass"` and the `"style"` conditions, as Sass can also
+use the CSS exports exposed through `"style"`. While in practice, `"style"`
+tends to be used solely for `css` files, we will support `scss`, `sass` and
+`css` files for either `"sass"` or `"style"`.
+
+## Types
+
+```ts
+import '../spec/js-api';
+```
+
+#### `usePkgImporter`
+
+If true, the compiler will use the built-in package importer to resolve any url
+with the `pkg` scheme. This importer follows Node logic to locate Sass files.
+
+Defaults to false.
+
+```ts
+declare module '../spec/js-api/options' {
+  interface Options<sync extends 'sync' | 'async'> {
+    /**
+     * Whether or not to enable the built-in package importer to resolve any url
+     * with the `pkg` scheme. This importer follows Node.js resolution logic.
+     *
+     * @defaultValue `false`
+     * @category Input
+     */
+    usePkgImporter?: boolean;
+  }
+}
+```
+
+## Semantics
+
+### Package Importers
+
+This proposal defines the requirements for Package Importers written by users or
+provided by implementations. It is a type of [Filesystem Importer] and will
+handle non-canonical URLs:
+
+- with the scheme `pkg`
+- followed by a package name
+- optionally followed by a path, with path segments separated with a forward
+  slash.
+
+The package name will often be the first path segment, but the importer should
+take into account any conventions in the environment. For instance, Node
+supports scoped package names, which start with `@` followed by 2 path segments.
+
+[filesystem importer]: ../spec/modules.md#filesystem-importer
+
+### Node Specific Semantics
+
+The Node package importer is an [importer] with an associated absolute `pkg:`
+URL named `base`. When the Node package importer is invoked with a string named
+`string` and a [previous url] `previousUrl`:
+
+- Let `url` be the result of [parsing `string` as a URL][parsing a url] with
+  `base` as the base URL. If this returns a failure, throw that failure.
```

Comment:
I'm not entirely sure what the purpose of `base` is here. As I understand it, the only non-canonical URLs we want package importers to handle are `pkg:` URLs, but if we parse `string` with a base then we're effectively allowing relative URLs as well. But it's also unclear to me where `base` is coming from or why we'd want it to be set.

Reply:
Accidental copy pasta- you're right this isn't needed.